Dockerfile

```diff
@@ -6,14 +6,18 @@ FROM docker.io/library/golang:1.17 as builder
 
 ARG release_tag
 
-# Copy the go source
-COPY . /workspace
-
 WORKDIR /workspace
 
-# cache deps before building and copying source so that we don't need to re-download as much
-# and so that source changes don't invalidate our downloaded layer
-RUN go mod download
+# Copy the Go Modules manifests
```

Comment:
All this is doing is making more layers. The COPY . /workspace isn't that big... Shouldn't we be able to just get away with removing the RUN go mod download?

Reply:
I believe on an `operator-sdk init` this is what you end up getting. ie only copy what you need, this more closely follows that. those layers get throw out though right? since this is the builder image and not the final image of the controller.

Reply:
In addition to what @acornett21 said, the codebase over time may have more things that aren't related to the compilation. E.g. more manifests and descriptors.
And obviously this helps filter out large debris like binaries and index img dbs, even if those somehow make it into git.

Reply:
Ok. Thanks for the clarification.